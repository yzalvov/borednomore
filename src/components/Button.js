import React from "react"
import { TouchableOpacity } from "react-native"
import { LargeText } from "./LargeText"

export const Button = ({ title, onPress }) => (
  <TouchableOpacity
    onPress={onPress}
    activeOpacity={0.7}
    style={{
      backgroundColor: "black",
      alignItems: "center",
      paddingVertical: 10,
      borderRadius: 11,
    }}>
    <LargeText style={{ color: "white", fontSize: 20 }}>{title}</LargeText>
  </TouchableOpacity>
)
