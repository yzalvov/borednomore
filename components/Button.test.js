import React from "react"
import renderer from "react-test-renderer"
import { Button } from "./Button"

it("Button renders correctly across screens", () => {
  const tree = renderer.create(<Button />).toJSON()
  expect(tree).toMatchSnapshot()
})
