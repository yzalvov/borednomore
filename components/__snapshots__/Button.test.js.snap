// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly across screens 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 11,
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "600",
        "lineHeight": 32.199999999999996,
      }
    }
  />
</View>
`;
